src/Datasets/PluckPreprocessor.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pest\Datasets;
+
+use InvalidArgumentException;
+use Pest\Contracts\Dataset\Preprocessor;
+
+/**
+ * @internal
+ */
+final class PluckPreprocessor implements Preprocessor
+{
+    public function getKey(): string
+    {
+        return 'pluck';
+    }
+
+    public function process(array $dataset, mixed $value): array
+    {
+        if (!is_string($value)) {
+            throw new InvalidArgumentException('The pluck preprocessor should be given a string.');
+        }
+
+        // @phpstan-ignore-next-line
+        return array_map(fn (array|object $values) => $values[$value], $dataset);
```

Comment:
Would be good to have this work on objects and arrays.

Reply:
hmm, I think the dataset resolves to an array before calling preprocess functions, is there any way it can be an object?

also I guess you mean `$values` to be `$dataset`?

Reply:
Doesn't this run on each item in the dataset? So for example, an array of Eloquent models?

Reply:
Hmm, no, pluck runs on each dataset and extracts only the key you want

that's the same behaviour of collection()->pluck(), it iterates over collection items and extract a key

or am I missing something?

Reply:
so, to clarify my answer:

```php
$dataset = [
    ['first_name' => 'Fabio', 'last_name' => 'Ivona'],
    ['first_name' => 'Luke', 'last_name' => 'Downing'],
];


 array_map(fn (array|object $values) => $values[$value], $dataset); 
 // $values =  ['first_name' => 'Fabio', 'last_name' => 'Ivona'] on the first iteration
 // $values =  ['first_name' => 'Luke', 'last_name' => 'Downing'] on the second iteration

Reply:
But a dataset could also be 

```php
$dataset = [
    new User(['first_name' => 'Fabio', 'last_name' => 'Ivona']),
    new Use(['first_name' => 'Luke', 'last_name' => 'Downing']),
];
```

Reply:
In which case I'd want it to get the property `first_name` from the `User` object.

Reply:
hmm, you're right, was focusing only on array[] datasets and missed single values one

let's change that value to array|object